worker/src/webhooks/target-transport.ts

```diff
@@ -0,0 +1,140 @@
+
+
+
+// import * as dns from 'dns';
+// import { Pool, Agent, setGlobalDispatcher } from 'undici'
+// import { Writable } from 'stream'
+
+// const agent = new Agent({
+//   connect: {
+//     lookup: dns.lookup,
+//   },
+// });
+
+// setGlobalDispatcher(agent)
+
+
+// const client = !hasBridgeTarget ? undefined : new Pool(process.env.LANGFUSE_WEBHOOKS_BRIDGE_TARGET_BASE_URL as string, {
+//   maxRedirections: 1,
+// })
+
+// interface RequestOptions {
+//   body: string
+// }
+
+// class SimpleRequest {
+//   private readonly dst: Writable
+//   private statusCode?: number
+
+//   reject: (err: Error) => void
+
+//   constructor(resolve: (value: unknown) => void, reject: (err: Error) => void) {
+//     this.dst = new Writable({
+//       write(chunk, encoding, callback) {
+//         callback()
+//       }
+//     }).on('finish', resolve)
+
+//     this.reject = (err: Error) => {
+//       reject(err)
+//     }
+//   }
+
+//   onConnect(_: any) { }
+
+//   onHeaders(statusCode: number, headers: Buffer[], resume: () => void, statusText: string) {
+//     this.statusCode = statusCode
+//     this.dst.on('drain', resume)
+//     return true
+//   }
+
+//   onData(chunk: string | Buffer) {
+//     return this.dst.write(chunk)
+//   }
+
+//   onComplete() {
+//     this.dst.end()
+//   }
+
+//   onError(err: Error) {
+//     this.reject(err)
+//   }
+// }
+
+
+// return new Promise((resolve, reject) => {
+//   client.dispatch({
+//     path: '/',
+//     method: 'POST',
+//     headers: {
+//       'content-type': 'application/json'
+//     },
+//     body: options.body
+//   }, new SimpleRequest(resolve, reject))
+// })
+
```

Comment:
NOTE: this is left here to give an impression what a better implementation could look like.

The node version will give undici fetch anyways, but is by far the slowest implementation https://github.com/nodejs/undici?tab=readme-ov-file#benchmarks

The problem is not the performance, but the long runs affecting the overall event loop